plugin/core/protocol.py

```diff
@@ -5840,8 +5840,10 @@ class TokenFormat(Enum):
 
 # AUTOGENERATED TYPES END
 
+R = TypeVar('R')
 
-class Request:
+
+class Request(Generic[R]):
```

Comment:
This is not strictly necessary but I was meaning to do this for a while. It allows us to type the responses.